opm/input/eclipse/Schedule/Well/WellKeywordHandlers.cpp

```diff
@@ -121,6 +121,7 @@ void handleWCONHIST(HandlerContext& handlerContext)
                                                             ParserKeywords::FBHPDEF::TARGET_BHP::defaultValue);
             }
 
+            properties->whistctl_cmode = handlerContext.state().whistctl();
```

Comment:
I've tried to read the development history here, but it's still not clear to me what problem this statement (`whistctl_cmode = state().whistctl()`) is supposed to solve.  **In principle**, the `whistctl_cmode` data member is supposed to be correctly handled in the [`make_shared<>()`](https://github.com/OPM/opm-common/blob/d809cb760c64e83ead7f545af34698b21ca898b8/opm/input/eclipse/Schedule/Well/WellKeywordHandlers.cpp#L102) call above (i.e., through the copy constructor for `WellProductionProperties`) and, moreover, properly updated in `handleWHISTCTL()`.  This statement, however, suggests that we're somehow missing an update somewhere and that could mean that there's a deeper problem hiding.

Reply:
> I _think_ the purpose was to pick up the case where a well happens to be injecting at the restart time, but later switches to production. In this case the well will not get any WellProductionProperties with the correct whistctl_cmode?

Ah, right.  Yes, that makes sense.  Would you mind adding a comment here to explain this reasoning, please?  That way we reduce the risk of this statement being removed because of someone using the same logic as me.  We **might** even consider putting the assignment behind a [`switching_from_injector`](https://github.com/OPM/opm-common/blob/333d25eae886a414faa2d730f29da0600bc55f3a/opm/input/eclipse/Schedule/Well/WellKeywordHandlers.cpp#L101) conditional as well.

Reply:
Good point, should have done that in the first place since I obviously had trouble remembering it myself.. :)  Updated now.